docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+|-----------------|---------------|------------------------------------------------|
+| amount          | double        | The budget amount to be deducted               |
+| category        | Category      | The category of budget to be deducted          |
+| date            | YearMonth     | The specific month and year of budget deducted |
+
+The BudgetBuddy class then calls the `execute()` method of the `DeductBudgetCommand` object which uses the 
+`deductAmount(category, amount)` method in the `Budget` class to deduct amount from the budget.
+
+The following method in the `BudgetManager` class called by `deductAmount(category, amount)` to delete a budget from 
+the `budgets` list if the total amount of the budget reaches zero:
+
+| Method               | Return Type | Relevance                                |
+|----------------------|-------------|------------------------------------------|
+| deleteBudget(amount) | void        | Delete budget from the list of `budgets` |      
+
+Finally, the result is displayed to the user using the `Ui` class `displayToUser()` method.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs
+for the Add Budget Feature:
+![DeductBudgetSequenceDiagram.drawio.png](diagrams/DeductBudgetSequenceDiagram.drawio.png)
+
+#### 4.4 List Budget Feature
+The List Budget feature enables users to view all existing budgets or filter them based on the date. This functionality 
+is controlled by the ListBudgetCommand class, which is produced by the Parser class based on user input. 
+The ListBudgetCommand class uses a ListBudgetValidator object to validate the provided date, checks if the list request 
+is valid, and, if valid, retrieves and displays the matching budgets through the UI. If the validation fails, 
+an error message is shown.
+
+#### 4.5 Search Expense Feature
+The Search Expense Feature enables users to search for specific expenses based on a description provided by the 
+user. This feature is managed by the `SearchExpensesCommand` class, initialized by the `Parser` class, with the help
+of a helper class `SearchExpenseValidator` to validate and extract the user description. 
+The `SearchExpensesCommand` object is then created with the keyword as an attribute. The class attributes and their
+relevance is as follows: 
+
+|Variable Name| Variable Type | Relevance |
+|-------------|---------------|-----------|
+|keyword| String | Description to find in expenses|
+
+The BudgetBuddy class then calls the `execute()` method of the `SearchExpenseCommand` object which uses the 
+`searchExpenses()` method in the `ExpenseManager` class, displaying the result to the user using the `Ui` class
+`displayToUser()` method.
+
+Below is a sequence diagram representing the execution of the Search Expense interaction: 
+![SearchExpenseSequenceDiagram.drawio.png](diagrams/SearchExpenseSequenceDiagram.drawio.png)
```

Comment:
Are alt and opt swapped? Recall alt should contain alternative paths and opt should contain a single optional path.
![image](https://github.com/user-attachments/assets/30aaf7c0-5e03-49c3-8bb0-1c722157bac8)